lib/dataloader/ecto.ex

```diff
@@ -786,6 +794,7 @@ if Code.ensure_loaded?(Ecto) do
         )
       end
 
+      # Fix me: never called because assoc is always length == 1
```

Comment:
I think you would hit this case with a `has_many, through: [:some_many_to_many_assoc]`. Eg in `test/support/user.ex` file, something like:


```elixir
    many_to_many(:liked_posts, Dataloader.Post, join_through: Dataloader.Like)
    has_many(:liked_posts_scores, through: [:liked_posts, :scores])
```

Reply:
Thanks a lot for the hint! I added a test for this in 3b17bac

I'll now adapt the `where` and `join_where` code to work with it.

Reply:
To clarify: the added test only tests this case:

```elixir
      defp build_preload_lateral_query(
             [%Ecto.Association.ManyToMany{} = assoc],
             query,
             :join_last
           ) do
```

I'll see if I can come up with cases for the remaining untested cases.